exercises/retree/canonical-data.json

```diff
@@ -0,0 +1,66 @@
+{
+    "exercise": "retree",
+    "version": "1.0.0",
+    "cases": [
+      {
+        "description": "Empty tree",
+        "property": "treeFromTraversals",
+        "input": {
+          "preorder": [],
+          "inorder": []
+        },
+        "expected": {}
+      },
+      {
+        "description": "Tree with one item",
+        "property": "treeFromTraversals",
+        "input": {
+            "preorder": ["a"],
+            "inorder": ["a"]
+        },
+        "expected": {"v":"a", "l": {}, "r": {}}
+      },
+      {
+        "description": "Tree with many items",
+        "property": "treeFromTraversals",
+        "input": {
+            "preorder": ["a", "i", "x", "f", "r"],
+            "inorder": ["i", "a", "f", "x", "r"]
+        },
+        "expected": {"v": "a",
+            "l": {"v": "i", "l": {}, "r": {}},
+            "r": {"v": "x",
+                "l": {"v": "f", "l": {}, "r": {}},
+                "r": {"v": "r", "l": {}, "r": {}}
+            }
+        }
+      },
+      {
+        "description": "Reject traversals of different length",
+        "property": "treeFromTraversals",
+        "input": {
+          "preorder": ["a", "b"],
+          "inorder": ["b", "a", "r"]
+        },
+        "expected": {"error": "traversals must have the same length"}
+      },
+      {
+          "description": "Reject inconsistent traversals of same length",
```

Comment:
I think the indentation of this test case is off (too much to the right).

Reply:
Other than that and we're ready to merge.

We should have a linting rule for this.

Reply:
> We should have a linting rule for this.

Yes, we should!